documentation/logging.markdown

```diff
@@ -0,0 +1,91 @@
+---
+title: "PuppetDB 4.2 » Logging Configuration"
+layout: default
+canonical: "/puppetdb/latest/logging.html
+---
+
+Structured logging
+-----
+
+PuppetDB uses logback, a standard Java logging library. But in certain
+subsystems, namely HA, we provide extended structured logging information. By
+configuring logback appropriately, you get JSON-formatted log messages with
+event-specific fields in each message.
+
+## Common fields
+
+If you use the recommended logger configuration, as described below, you will
+see the following fields in each JSON log message:
+
+* @timestamp
+* message
+* logger_name
+* thread_name
+* level
+* level_value (numeric, suitable for sorting)
+* stack_trace
+
+Additional relevant fields may be added in any given message, but this base set
+will always be present.
+
+## JSON text
+
+If you want to log JSON data where you would otherwise log regular text, replace the 'encoder' element in
+your logback.xml with this one:
+
+    <encoder class="net.logstash.logback.encoder.LoggingEventCompositeJsonEncoder">
+      <providers>
+        <timestamp/>
+        <message/>
+        <loggerName/>
+        <threadName/>
+        <logLevel/>
+        <logLevelValue/>
+        <stackTrace/>
+        <logstashMarkers/>
+      </providers>
+    </encoder>
+
+Even though this says 'logstash' on it, it works completely independently from
+any log aggregation system. The final `<logstashMarkers/>` element is
+important - that inserts our custom properties into each json message.
```

Comment:
I don't think "is important" adds anything here. The final element <foo> inserts our custom properties.